Framework/Algorithms/src/ConvertSpectrumAxis2.cpp

```diff
@@ -153,13 +153,23 @@ void ConvertSpectrumAxis2::createElasticQMap(API::Progress &progress,
   else if (emodeStr == "Indirect")
     emode = 2;
 
-  auto &spectrumInfo = inputWS->spectrumInfo();
+  const auto &spectrumInfo = inputWS->spectrumInfo();
+  const auto &detectorInfo = inputWS->detectorInfo();
   for (size_t i = 0; i < nHist; i++) {
     double theta(0.0), efixed(0.0);
     if (!spectrumInfo.isMonitor(i)) {
```

Comment:
There seems to be an assumption here that a detector exists for a spectrum. Should there be a check? Perhaps not as there was not one before.